SRTNet.cpp

```diff
@@ -447,56 +584,77 @@ bool SRTNet::startClient(const std::string& host,
-    std::stringstream portAsString;
-    portAsString << port;
-    result = getaddrinfo(host.c_str(), portAsString.str().c_str(), &hints, &svr);
-    if (result) {
-        SRT_LOGGER(true, LOGG_FATAL,
-                   "Failed getting the IP target for > " << host << ":" << port << " Errno: " << result);
-        return false;
-    }
-
-    SRT_LOGGER(true, LOGG_NOTIFY, "SRT connect");
-    for (hld = svr; hld; hld = hld->ai_next) {
-        result = srt_connect(mContext, reinterpret_cast<sockaddr*>(hld->ai_addr), hld->ai_addrlen);
-        if (result != SRT_ERROR) {
-            SRT_LOGGER(true, LOGG_NOTIFY, "Connected to SRT Server " << std::endl)
-            break;
-        }
-    }
-    if (result == SRT_ERROR) {
-        srt_close(mContext);
-        freeaddrinfo(svr);
-        SRT_LOGGER(true, LOGG_FATAL, "srt_connect failed " << std::endl);
-        return false;
-    }
-    freeaddrinfo(svr);
-    mCurrentMode = Mode::client;
-    mClientActive = true;
-    mWorkerThread = std::thread(&SRTNet::clientWorker, this);
     return true;
 }
 
 void SRTNet::clientWorker() {
     while (mClientActive) {
+        if (!mClientConnected) {
+            // Get all remote addresses for connection
+            struct addrinfo hints = {0};
+            struct addrinfo* svr;
+            struct addrinfo* hld;
+            hints.ai_socktype = SOCK_DGRAM;
+            hints.ai_protocol = IPPROTO_UDP;
+            hints.ai_family = AF_UNSPEC;
+            std::stringstream portAsString;
+            portAsString << mConfiguration.mRemotePort;
+            int result = getaddrinfo(mConfiguration.mRemoteHost.c_str(), portAsString.str().c_str(), &hints, &svr);
+            if (result) {
+                SRT_LOGGER(true, LOGG_FATAL,
+                        "Failed getting the IP target for > " << mConfiguration.mRemoteHost << ":" << mConfiguration.mRemotePort << " Errno: " << result);
+                break;
+            }
+
+            for (hld = svr; hld; hld = hld->ai_next) {
+                result = srt_connect(mContext, reinterpret_cast<sockaddr*>(hld->ai_addr), hld->ai_addrlen);
+                if (result != SRT_ERROR) {
+                    SRT_LOGGER(true, LOGG_NOTIFY, "Connected to SRT Server " << std::endl)
+                    mClientConnected = true;
+                    // Break for-loop on first successful connect call
+                    break;
+                }
+            }
+            freeaddrinfo(svr);
+            if (result == SRT_ERROR) {
+                // Failed to connect caller/client, try again.
+                int rejectReason = srt_getrejectreason(mContext);
+                if (rejectReason != SRT_REJECT_REASON::SRT_REJ_TIMEOUT) {
+                    if (rejectReason == SRT_REJECT_REASON::SRT_REJ_BADSECRET ||
+                        rejectReason == SRT_REJECT_REASON::SRT_REJ_UNSECURE) {
+                        SRT_LOGGER(true, LOGG_FATAL, "srt_connect failed (bad PSK):" << srt_getlasterror_str() << std::endl);
```

Comment:
Why try again, will it ever connect if we have a bad PSK?

Reply:
We discussed this with Mikael, and he thinks the SRT caller should keep trying to connect in case the receiving side changed the PSK.